test/integration/model/upsert.test.js

```diff
@@ -487,6 +487,34 @@ describe(Support.getTestDialectTeaser('Model'), () => {
             });
         });
       }
+
+      if (dialect.match(/^postgres/)) {
+        it('works when deletedAt is Infinity and part of primary key', function() {
+          const User = this.sequelize.define('User', {
+            name: {
+              type: DataTypes.STRING,
+              primaryKey: true
+            },
+            address: DataTypes.STRING,
+            deletedAt: {
+              type: DataTypes.DATE,
+              primaryKey: true,
+              allowNull: false,
+              defaultValue: Infinity
+            }
+          }, {
+            paranoid: true
+          });
+
+          return User.sync({ force: true }).then(() => {
+            return User.build({ name: 'user1' })
+              .save()
+              .then(() => {
+                return User.upsert({ name: 'user1', address: 'address' });
```

Comment:
Test should be more expressive. Add three records (dont use build, just use Promise.all / create)

1) No deletedAt set
2) deletedAt set to Infinity
3) deletedAt set to -Infinity

assert 2 records are returned

Reply:
@sushantdhiman Updated. Let me know if that's what you had in mind.